src/plugins/mer/mergeneraloptionswidget.cpp

```diff
@@ -91,5 +186,39 @@ QString MerGeneralOptionsWidget::searchKeywords() const
     return keywords;
 }
 
+void MerGeneralOptionsWidget::on_addProfileButton_clicked()
+{
+    bool ok;
+    const QString name = QInputDialog::getText(this, tr("Create device model"),
+                                               tr("Provide unique device model name:"), QLineEdit::Normal,
+                                               tr("Custom"), &ok);
+    if (!ok || name.isEmpty())
+        return;
+
+    if (MerSettings::addDeviceModel(name)) {
+        m_ui->deviceModelComboBox->selectDeviceModel(name);
+    } else {
+        QMessageBox::warning(this, tr("Create device model warning"), tr("Emulator device model is not created.\n"
+                                                                    "Provided name \"%1\" is not unique.").arg(name));
+    }
+}
+
+void MerGeneralOptionsWidget::on_removeProfileButton_clicked()
+{
+    const auto name = m_ui->deviceModelComboBox->currentData().toString();
+    const int ret = QMessageBox::question(this, tr("Delete device model"),
+                                          tr("This action will completely delete device model \"%1\".\n"
+                                             "Do you want to proceed?").arg(name),
```

Comment:
I checked how QtCreator handles removing something in Options window, and think I should omit QMessageBox because changes will not apply until user press Ok button.

Reply:
Good idea :)

Reply:
QMessageBox::question removed.